ee/codegen/src/generators/workflow.ts

```diff
@@ -636,14 +663,161 @@ export class Workflow {
     });
   }
 
+  private buildGraphAttribute(): AstNode | undefined {
+    if (!this.breadthFirstGraphGeneration) {
+      return this.buildGraphLegacy({
+        portContexts: this.entrypointPortContexts,
+      });
+    }
+
+    // This graph attribute generation is a breadth-first traversal of the WorkflowEdges
+    // starting at the entrypoint port edges. We process each one sequentially and incrementally
+    // build up the graph as we go.
```

Comment:
I'm defining all types and helper methods I need internal to this method for now, and happy to split out and reorg when ready. Also happy to reorg now if there are strong opinions